### ISSUE TITLE
ip-1.1.2 build failure with GHC 8.4

### ISSUE THREAD

**Issue by DanBurton** [andrewthad/haskell-ip#34]
Add semigroup instances
See also: https://ghc.haskell.org/trac/ghc/wiki/Migration/8.4#SemigroupMonoidsuperclasses

```
[ 2 of 11] Compiling Data.Text.Builder.Variable ( src/Data/Text/Builder/Variable.hs, .stack-work/dist/x86_64-osx/Cabal-2.2.0.0/build/Data/Text/Builder/Variable.o )

/Users/dan/scratch/ip-1.1.2/src/Data/Text/Builder/Variable.hs:41:10: error:
    • No instance for (Semigroup (Builder a))
        arising from the superclasses of an instance declaration
    • In the instance declaration for ‘Monoid (Builder a)’
   |
41 | instance Monoid (Builder a) where
   |          ^^^^^^^^^^^^^^^^^^
```

I was able to reproduce this locally like so:

```bash
stack unpack ip-1.1.2 && cd ip-1.1.2
echo 'resolver: nightly-2018-03-21' > stack.yaml
stack build
```

**Comment by andrewthad** [comment]
Should be fixed with 0a9d76d491ad814109b230f9eb38d1c85c795127. I just released this to hackage as a part of `ip-1.2.0`.

**Comment by DanBurton** [comment]
👍
